README.md

```diff
@@ -19,71 +19,151 @@ ubuntu1804-64 | :white_check_mark:
 ubuntu2004-64 | :white_check_mark:
 win12-64 | :white_check_mark:
 
-## Execute
+It's possible to run this test suite for all the OSs in the support matrix above, or for a subset of those OSs. The possible use cases are:
 
-First, you need to bring the machines up:
+- test all Beats for all OSs
+- test all Beats for only one OS
+- test one Beat for all OSs
+- test one Beat for only one OS
+- test all Beats in batches for all OSs
+- test all Beats in batches for one group of OSs
 
-    make setup
+> The supported groups are 'centos', 'debian', 'sles' and 'windows'.
 
-Then you can use Ansible to run the tests. Because they involve lots of VMs and
-commands executed over SSH, these tests are slow (currently 15 minutes in
-total). However, while creating tests or checking something quickly, you can use
-different Ansible commands to execute only a subset.
+## Preparing the machines
+
+### For all OSs
+
+You need to bring all the machines up:
+
+```shell
+make setup
+```
+
+> If the command fails with the following error, please retry `make setup` again:
+
+```
+The following SSH command responded with a non-zero exit status.
+Vagrant assumes that this means the command failed!
+
+umount /mnt
+
+Stdout from the command:
+
+
+
+Stderr from the command:
+
+umount: /mnt: not mounted
+```
+
+### For one OS
+
+Only a particular OS will be spun by Vagrant. In the following example, we are going to provision a Debian 8 amd64 image:
+
+```shell
+MACHINE=tester-debian8-64 make setup
+```
+
+> You will find the machine types in the [Vagrantfile](./Vagrantfile), under the `config.vm.define` attribute. The name convention uses `tester-` as prefix of the OS listed in the above support matrix.
+
+## Running the tests
+
+Once the target machines are ready, you need Ansible to run the tests. Because they could involve lots of VMs and commands executed over SSH, these tests are slow (currently 15 minutes in total).
+
+The commands that are available for running the tests are:
+
+- `make run`: it will run the tests for all the OSs in the support matrix.
+- `make run-group`: it will run the tests for all the OSs in a group of OSs. For each ansible group, it will start the vm, check the VM's status, configure SSH, run the ansible playbook, and finally will destroy the vm.
+- `make batch`: it will run the tests in batches. Since it processes each group sequentially, it uses less CPU and memory.
```

Comment:
What do you think to add those details to the Make goals? And then provide a `help` make goal that provide those details on the fly?